File: src/primes.pl
#!/usr/bin/perl

use strict;
use warnings;
use feature 'say';

my $MAX = 1_000_000_000; # 1 billion
my $MAX_SQRT = 31623;

my @nums;
$#nums = $MAX;

# 0 and 1 are not prime
$nums[0] = $nums[1] = 1;

# Computing prime numbers
for my $i (2 .. $MAX_SQRT) {
    next if $nums[$i];
    for (my $j = $i+$i; $j < $MAX; $j += $i) {
        $nums[$j] = 1;
    }
}

# Counting prime numbers
my $total = 0;
$total += !$_ for @nums;
say $total

File: src/primes.R
#!/usr/bin/env Rscript
# WARNING: R is 1-indexed
printf <- function(...) cat(sprintf(...))
MAX = as.integer(commandArgs(trailingOnly = TRUE)[1])

nums = vector(length = MAX)

# 0 and 1 are not prime, 
nums[c(1,2)] = TRUE

# Computing prime numbers
for (i in 3:floor(sqrt(MAX)+1)) {
  if (nums[i] == TRUE) {
    next
  }
  nums[seq(i*2-1, MAX, i-1)] = TRUE
}

total = MAX - sum(nums)
printf("%d\n", total)

